python/semantic_kernel/events/kernel_events_args.py

```diff
@@ -1,31 +1,42 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-from semantic_kernel.orchestration.kernel_context import KernelContext
-from semantic_kernel.plugin_definition.function_view import FunctionView
+from pydantic import Field
 
+from semantic_kernel.functions.kernel_arguments import KernelArguments
+from semantic_kernel.functions.kernel_function_metadata import KernelFunctionMetadata
+from semantic_kernel.kernel_pydantic import KernelBaseModel
 
-class KernelEventArgs:
-    def __init__(self, function_view: FunctionView, context: KernelContext):
-        if context is None or function_view is None:
-            raise ValueError("function_view and context cannot be None")
 
-        self._function_view = function_view
-        self._context = context
+class KernelEventArgs(KernelBaseModel):
+    """Base class for Kernel Event args.
 
-        # Temporal cancellationToken to sync C#
-        self._is_cancel_requested = False
+    Receives relevant parts of the the execution, either before (invoking) or after (invoked) the function is executed.
+    When a handler changes the arguments in the invoking event,
+    the new arguments are passed to the invoked event,
+    make sure to use the update_arguments function, since that also raises the flag that the arguments were updated.
 
-    @property
-    def function_view(self):
-        return self._function_view
+    Args:
+        function_view (FunctionView): The function that is being executed.
+        arguments (KernelArguments): The arguments that are being passed to the function.
 
-    @property
-    def context(self):
-        return self._context
+    Flags:
+        updated_arguments (bool): Whether the arguments were updated, default False.
+        is_cancel_requested (bool): Whether the function execution has to be canceled, default False.
 
-    @property
-    def is_cancel_requested(self):
-        return self._is_cancel_requested
+    Methods:
+        cancel: Sets the is_cancel_requested flag to True.
+        update_arguments: Updates the arguments and raises the updated_arguments flag.
+
+    """
+
+    function_view: KernelFunctionMetadata
```

Comment:
Are we leaving the name as `function_view`? Or will we update to something more related to the kernel function metadata?